enterprise/api/v1/views/enterprise_customer_sso_configuration.py

```diff
@@ -59,3 +78,115 @@ def oauth_orchestration_complete(self, request, configuration_uuid, *args, **kwa
+
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, *args, **kwargs: get_customer_from_request(request)
+    )
+    def retrieve(self, request, *args, **kwargs):
+        configuration_uuid = kwargs.get('configuration_uuid')
+        # Look for the configuration record
+        sso_configuration_record = self.queryset.filter(pk=configuration_uuid).first()
+        if not sso_configuration_record:
+            return Response(status=HTTP_404_NOT_FOUND)
+        # Make sure the requesting user is part of the enterprise customer associated with the configuration record
+        # or are a staff member
+        config_customer = sso_configuration_record.enterprise_customer
+        if not check_user_part_of_customer(request.user, config_customer):
+            return Response(status=HTTP_404_NOT_FOUND)
+
+        serializer = self.serializer_class(sso_configuration_record)
+        return Response(serializer.data)
+
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, *args, **kwargs: get_customer_from_request(request)
+    )
+    def list(self, request, *args, **kwargs):
+        list_queryset = self.queryset
+
+        # If the request includes a customer uuid, we require through permissioning that the requesting user is an
+        # admin of that customer
+        if provided_customer := request.query_params.get('enterprise_customer'):
+            enterprise_customer = EnterpriseCustomer.objects.filter(pk=provided_customer)
+            list_queryset = list_queryset.filter(enterprise_customer__in=enterprise_customer)
+
+        # For non-staff users, only show the configurations that are connected to enterprise customers
+        # that they are associated with
+        if not request.user.is_staff:
+            ent_customers = EnterpriseCustomer.objects.filter(
+                enterprise_customer_users__in=EnterpriseCustomerUser.objects.filter(user_id=request.user.id)
+            )
+            list_queryset = list_queryset.filter(enterprise_customer__in=ent_customers)
+
+        serializer = self.serializer_class(list_queryset, many=True)
+        return Response(serializer.data, status=HTTP_200_OK)
+
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, *args, **kwargs: get_customer_from_request(request)
+    )
+    def create(self, request, *args, **kwargs):
+        # Force the enterprise customer to be the one associated with the user
+        request_data = request.data.dict().copy()
+        requesting_user_customer = request_data.get('enterprise_customer')
+        if requesting_user_customer:
+            try:
+                enterprise_customer = EnterpriseCustomer.objects.get(uuid=requesting_user_customer)
+            except EnterpriseCustomer.DoesNotExist:
+                return Response({'error': 'Must provide valid enterprise customer'}, status=HTTP_400_BAD_REQUEST)
```

Comment:
Same response text is used in multiple places here.

Reply:
this was done as a minor security issue in exposing customer IDs, I wanted to keep the message the same if the user requests a customer that doesn't exist and a customer that the user doesn't belong to

Reply:
That part makes sense, but we can at least put the common error message string in a central location.